horizomer/utils/tree.py

```diff
@@ -15,6 +15,59 @@
 from skbio import TreeNode
 
 
+def support(node):
+    """Get support value of a node.
+
+    Parameters
+    ----------
+    node : skbio.TreeNode
+        node to get support value of
+
+    Returns
+    -------
+    float or None
+        support value of the node, or None if not available
+
+    Notes
+    -----
+    A "support value" is defined as the numeric form of a whole node label
+    without ":", or the part preceding the first ":" in the node label.
+    - For examples: "(a,b)1.0", "(a,b)1.0:2.5", and "(a,b)'1.0:species_A'". In
+    these cases the support values are all 1.0.
+    - For examples: "(a,b):1.0" and "(a,b)species_A". In these cases there are
+    no support values.
+    """
+    try:
+        return float(node.name.split(':')[0])
+    except (ValueError, AttributeError):
+        return None
+
+
+def collapse(node):
+    """Collapse an internal node of a tree.
```

Comment:
I don't get what the functions purpose should be from reading the doc string. Can you give an example?

Reply:
Am I correct to say that children are moved up to the parent while self get's deleted?
What is the usecase? Do you want to remove internal nodes with just once child?

Reply:
The usecase is that one wants to collapse nodes with low support value or short branch length, two signs of unreliable topology. The two other functions are coded for this purpose. For example in this tree:
```
                    /-a
          /c-------|
         |          \-b
---------|
         |          /-d
          \f-------|
                    \-e
```
Assume node `c` has a very low support value. We cannot trust this topology. If the tree is fed to downstream applications like UniFrac, this unreliability may cause problem. So we want to collapse this node, and the topology will become:
```
          /-a
         |
---------|--b
         |
         |          /-d
          \f-------|
                    \-e
```

Reply:
I think you should add this beautiful example to the documentation.
And maybe come up with another name like "lift_children" ?

Reply:
Thanks! Sure, I can add it. I think "collapse" is the standard term in phylogenetics.

![screen shot 2018-01-26 at 3 22 43 pm](https://user-images.githubusercontent.com/4396343/35464798-d0ced15a-02ac-11e8-9e0f-c1f4fdfbe4ea.png)

Reply:
hm the figtree example shows that "collapse" is merging a sub-tree into a single node or nodelike object. You thus gave a counter example LOL

Reply:
Well... if you just google "phylogenetic tree collapse node" you will get lots of hits. Like this [one](http://treegraph.bioinfweb.info/Help/wiki/Collapsing_nodes_by_support)

Reply:
Agree with @sjanssen2,  collapse can refer to a wide-range of possible operations, including aggregating abundances on tips, merging together nodes into a single node, or creating "super" nodes.

At the very least, I'd suggest inlining that specific example in the documentation.  Renaming the function to `collapse_support` may also help.

Reply:
Hi @mortonjt @sjanssen2 how about we do `collapse_node`?

Reply:
I'd suggest that the fact that this is stressing our intuitive understanding of the word is indicative of a problem with the terminology, and we should consider that. 

Noodling a bit on this, I think the problem stems from *which* node is being considered here. When we talk about things *collapsing* we usually imagine something larger being condensed into the same thing, only smaller: a star -> a black hole; a building -> a pile of rubble; a cardboard box -> folded cardboard. 

The FigTree usage fits this intution: the *node itself* remains in existence, but the *internal structure* of the node is destroyed. 

The other usage actually destroys the node entirely: it ceases to exist, and the internal structure (the set of descending branches) is removed and separated into no-longer-especially-connected parts. Verbs that more effectively convey this sort of change might be: dissolve, unpack, discombobulate, disaggregate, explode, dissect. 

I'd lean toward **collapse** referring to the maintenance of the node itself, but the dissolution of all internal structure (i.e. all sub-nodes) into one polytomy (or alternative object -- with the recursive node structure this is maybe a less useful operation than in the visual model); and **dissolve** referring to the behavior here, where the immediate sub-branches are grafted onto the parent node. Dissolve might be a bit too strong -- it conveys a *lot* of structural loss, so **disaggregate** or **unpack** might be more apt.

Reply:
Hi Jon, thanks for the very detailed opinion. I explored a little bit and felt that the terminology is ambiguous. [DendroPy](https://pythonhosted.org/DendroPy/library/treemodel.html?highlight=collapse#dendropy.datamodel.treemodel.Edge.collapse), [Mesquite](https://mesquiteproject.wikispaces.com/Trees) and [TG2](http://treegraph.bioinfweb.info/Help/wiki/Collapsing_nodes_by_support) performs "collapse" in the matter I am doing. Whereas FigTree, Dendroscope and [ETE3](https://github.com/etetoolkit/ete/blob/master/ete3/phylo/phylotree.py#L715) appear to "collapse" an entire clade.

My feeling is that programs that actually manipulate tree topology do "collapse" by deleting a node and raising child nodes, while tree display programs convert a clade into a tip.

My overall suggestion is that we don't invent new terms, but adopt an existing one, i.e., **collapse**, even if its meaning is not crystal clear. What do you think?

Reply:
@sjanssen2 @mortonjt @tanaes I still believe that **collapse** is the most proper term here, especially comparing to English verbs that have not been frequently used by phylogeneticists. However, I do not have strong opinion in terminology, and am willing to make compromise.

Therefore I am proposing to use the function name `collapse_node` here, to describe the action of removing a node and elevating its children, while leaving another function `collapse_clade` for the future, which will convert an entire clade to a single tip. I think these two operations have summarized the two categories of examples I listed above.